### DISCUSSION TITLE
Support for adding `line_item` with metadata

### DISCUSSION THREAD

**Discussion by guyathomas** [medusajs/medusa#4845] [Feature Requests]
Since `LineItem` supports metadata, I'd like to propose that we add support for including that.

* `addItem` and `addToCart` should support specifying of metadata
* support in the `create-line-item` route.

I'm open to adding this support in myself, but wanted to vet the idea.

**Comment by srindom** [comment]
This is already supported. You should be able to do:

```
POST /store/carts/:id/line-items
{
  quantity: 1,
  variant_id: "variant_1234",
  metadata: { my: "metadata" }
}
```
